mantidproject/mantid
# Create Inital Focusing Tab for New Engineering Diffraction GUI

**Description of work.**
A focusing tab has been added to allow for focused data to be generated from the run files. It takes the values from the calibration tab and uses them while running the focusing algorithm. 
It also saves the outputs to the users file system in the same location as the calibration output files (this location cannot be changed for now). 

**To test:**
Note: You may need to add `Diffraction/Engineering_Diffraction_2.py` to `Mantid.properties.template`
1. Ensure that the GUI opens properly in both plot and workbench.
2. Attempt a focus without doing a calibration. This should fail.
3. Perform a calibration and then attempt a focus. V: 307521 C: 305738 Focus: 305761
4. Check that a plot appears if the check box is ticked.
5. Check that only the ticked banks have workspaces and files created. 

Blocked By #27046.

Fixes #27012.

*This does not require release notes* because **the GUI is not yet in a state where users would find it useful.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
